application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -94,54 +99,121 @@ public BotCore(JDA jda, Database database, Config config) {
             .map(UserInteractor.class::cast)
-            .collect(Collectors.toMap(UserInteractor::getName, Function.identity()));
+            .filter(validateInteractor())
+            .toList();
+
+        nameToInteractor = interactors.stream().collect(Collectors.toMap(interactor -> {
+            String name = interactor.getName();
+
+            if (interactor instanceof SlashCommand) {
+                return "s-" + name;
+            } else if (interactor instanceof MessageContextCommand) {
+                return "mc-" + name;
+            } else if (interactor instanceof UserContextCommand) {
+                return "uc-" + name;
+            }
+
+            return name;
+        }, Function.identity()));
 
-        // Reload Command
-        if (nameToInteractor.containsKey(RELOAD_COMMAND)) {
-            throw new IllegalStateException(
-                    "The 'reload' command is a special reserved command that must not be used by other user interactors");
-        }
-        nameToInteractor.put(RELOAD_COMMAND, new ReloadCommand(this));
 
         // Component Id Store
         componentIdStore = new ComponentIdStore(database);
         componentIdStore.addComponentIdRemovedListener(BotCore::onComponentIdRemoved);
         componentIdParser = uuid -> componentIdStore.get(UUID.fromString(uuid));
-        nameToInteractor.values()
-            .forEach(slashCommand -> slashCommand
-                .acceptComponentIdGenerator(((componentId, lifespan) -> {
-                    UUID uuid = UUID.randomUUID();
-                    componentIdStore.putOrThrow(uuid, componentId, lifespan);
-                    return uuid.toString();
-                })));
+        interactors.forEach(slashCommand -> slashCommand
+            .acceptComponentIdGenerator(((componentId, lifespan) -> {
+                UUID uuid = UUID.randomUUID();
+                componentIdStore.putOrThrow(uuid, componentId, lifespan);
+                return uuid.toString();
+            })));
 
         if (logger.isInfoEnabled()) {
-            logger.info("Available user interactors: {}", nameToInteractor.keySet());
+            logger.info("Available user interactors: {}", interactors);
         }
     }
 
+    /**
+     * Validates the given interactor
+     *
+     * @return whenever the command should be checked
+     */
+    @NotNull
+    private static Predicate<UserInteractor> validateInteractor() {
+        return interactor -> {
+            String name = interactor.getName();
+
+            if (name == null) {
+                logger.error("An interactor's name shouldn't be null! {}", interactor);
+                return false;
+            }
+
+            if (name.startsWith("s-") || name.startsWith("mc-") || name.startsWith("uc-")) {
+                logger.warn(
+                        "Names that starts with s-, mc- or uc- aren't recommended and might break other commands. (name: {})",
+                        name);
```

Comment:
im voting for fail-fast here. we must not create any command with that name. instead of being defensive here, be aggressive and fail the code